src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -901,25 +909,30 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
         // for the platform. Currently musl and arm64 should be only cases to use
         // this method to determine cache size.
         //
-        size_t size;
-
-        if (ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index0/size", &size))
-            cacheSize = std::max(cacheSize, size);
-        if (ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index1/size", &size))
-            cacheSize = std::max(cacheSize, size);
-        if (ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index2/size", &size))
-            cacheSize = std::max(cacheSize, size);
-        if (ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index3/size", &size))
-            cacheSize = std::max(cacheSize, size);
-        if (ReadMemoryValueFromFile("/sys/devices/system/cpu/cpu0/cache/index4/size", &size))
-            cacheSize = std::max(cacheSize, size);
+        size_t level;
+        for (int i = 0; i < 5; i++)
+        {
+            string path_to_index = "/sys/devices/system/cpu/cpu0/cache/index" + to_string(i);
+            if (ReadMemoryValueFromFile((path_to_index + "/size").c_str(), &size))
+            {
+                if (ReadMemoryValueFromFile((path_to_index + "/level").c_str(), &level))
+                {
+                    CHECK_CACHE_SIZE(level)
+                }
+                else
+                {
+                    cacheSize = std::max(cacheSize, size);
+                }
+            }
+        }
     }
 #endif
 
-#if defined(HOST_ARM64) && !defined(TARGET_OSX)
-    if (cacheSize == 0)
+#if (defined(HOST_ARM64) || defined(HOST_LOONGARCH64)) && !defined(TARGET_OSX)
+    if ((cacheSize == 0) || (cacheLevel != 3))
```

Comment:
>    if ((cacheSize == 0)  [](http://example.com/codeflow?start=0&length=25)

@janvorli do you know when we would get 0 for cacheSize here? have you come across any such cases?

Reply:
For example, on my arm64 Linux device, there is no `/sys/devices/system/cpu/cpu0/cache` folder and all of the `sysconf` above return 0.

Reply:
And the distro is Ubuntu 18.04